Update User Guide - Visual Style Editor (#10028)
geosolutions-it/MapStore2: docs/user-guide/layer-settings.md

<img src="../img/layer-settings/classification_style_ex.jpg" class="ms-docimage">

!!!note
    The *Classification style* method is also available for **WFS** and **Vector** layers.

!!!note

>     From the *Classification style* options, it is possible to activate the **Thematic Layer** option thought the <img src="../img/button/configute_lite_button.jpg" class="ms-docbutton"/> button. 

Reviewer: ```suggestion
    From the *Classification style* options, it is also possible to manage **Custom Parameter** thought the <img src="../img/button/configute_lite_button.jpg" class="ms-docbutton"/> button. That's really useful to style [SQL Views](https://docs.geoserver.org/latest/en/user/data/database/sqlview.html) defined in GeoServer by using one of the available _viewparams_
```

>     <img src="../img/layer-settings/configure_popup.jpg" class="ms-docimage" style="max-width:400px;">

Reviewer: Use this. Optimized it for the web, please.
![image](https://github.com/geosolutions-it/MapStore2/assets/1280027/05f4587e-ca03-4657-addb-537c19f65a6d)

>     Here the user can add custom parameters that will be visible in the *Visual Style Editor*, above the *Color Palette* option. The *Thematic Layer* property provides an array of fields where each one represents a select input, once a user selects one of this input the associated view parameter is included in the classify.json request to the SLD service. An example can be the following one:

Reviewer: ```suggestion
    Here the user can add custom parameters that will be visible in the *Visual Style Editor*, above the *Color Palette* option. The *Custom Parameters* requires an array of fields to be configured according to the available viewparams defined in the SQL View. 
```

<img src="../img/layer-settings/classify_icon_ex.jpg" class="ms-docimage">

> #### Style with properties

Reviewer: ```suggestion
#### Style with property value
```

> [MapStore](https://mapstore.geosolutionsgroup.com/mapstore/#/) allows to choose a property of a layer's feature as the value for a symbolizer option. The *Style with properties* function is available for *Marker*, *Icon*, *Line*, *Fill*, *Text* and *3D model* by clicking on the <img src="../img/button/options_button.jpg" class="ms-docbutton"/> button of each symbolizer option supported by this function and choosing **Property value** option from the dropdown menu.

Reviewer: ```suggestion
[MapStore](https://mapstore.geosolutionsgroup.com/mapstore/#/) allows you to use a property of the layer feature as value for a symbolizer property. The *Property value* function is available for *Marker*, *Icon*, *Line*, *Fill*, *Text* and *3D model* by clicking on the <img src="../img/button/options_button.jpg" class="ms-docbutton"/> button of each symbolizer property supporting it.
```

<img src="../img/layer-settings/properties_style.jpg" class="ms-docimage"  style="max-width:500px;">

> By default, **Constant value** is selected. If the user chooses **Property value**, a drop-down menu appears so that the desired layer attribute can be selected to use the necessary values from the styling property.

Reviewer: ```suggestion
By default, **Constant value** is selected. If the user chooses **Property value**, a drop-down menu appears so that the desired layer attribute can be selected to use its values for the styling property.
```

<video class="ms-docimage" controls><source src="../img/layer-settings/property-value-example.mp4"></video>

!!!warning
    As a *Property value* only attributes of type `string` or `number` are currently supported.

!!!note

>     The *Style with properties* method is also available for **WFS** and **Vector** layers.

Reviewer: ```suggestion
    The *Style with property* method is available only for **WFS** and **Vector** layers.
```

Furthermore, for **WFS layers**, [MapStore](https://mapstore.geosolutionsgroup.com/mapstore/#/) adds some additional styling options in the *Visual Style Editor* such as:

> * **Bring to front** (available for Icon, Mark and 3D model symbolizers) to bring in front and so to make visible (if enable) all features covered by 3D Tile layers and the Terrain layer (for this last case when the *depth test against terrain* option is enabled in Global Settings).

Reviewer: ```suggestion
* **Bring to front** (available for Icon, Mark and 3D model symbolizers) to bring in front and so to make visible (if enabled) all features covered by 3D Tile layers and the Terrain layer (for this last case when the *depth test against terrain* option is enabled in Global Settings).
```

<img src="../img/layer-settings/polygon-type.jpg" class="ms-docimage">

> * **Extrusion Height** (available for Line and Fill symbolizers) to configure the height value of the symbol to extrude. It is also possible to enable/disable the **Extrusion relative to geometry** and, only for the *Line* symbolizers, the user can customize the **Color** and the **Type**, choosing between `Wall`, `Circle` and `Square` options, for the extruded features.

Reviewer: ```suggestion
* **Extrusion Height** (available for Line and Fill symbolizers) to configure the height value of the feature to be extruded. It is also possible to enable/disable the **Extrusion relative to geometry** (from the highest point of the geometry) and, only for the *Line* symbolizers, the user can customize the **Extrusion color** and the **Extrusion type**, choosing between `Wall`, `Circle` and `Square` options, for the extruded features.
```
